docs(fields): add missing validation methods in jsdoc and improve fields documentation (#958)
Refinitiv/refinitiv-ui: documents/src/pages/elements/password-field.md

`ef-password-field` has validation logic similar to a native input. When a user types the invalid value into the control, error style will be shown to notify the user. However, if the control is being initialised with an invalid value, `reportValidity()` must be called to ensure the error style is applied.

::
```javascript
::password-field::
const passwordField = document.querySelector('ef-password-field');
const button = document.querySelector("ef-button");

button.addEventListener('tap', () => {
  passwordField.reportValidity();
})
```
```html
<ef-password-field pattern="[a-z]" value="1"></ef-password-field>
<ef-button>Submit</ef-button>
```
::

### Displaying error messages

> Whenever input is invalid, the error attribute will be added to the element. You can use the `error` property to check if input is currently in the error state.

Reviewer: ```suggestion
Whenever input is invalid, the `error` attribute will be added to the element. You can use the `error` property to check if input is currently in the error state. Note that, if input is initialised with invalid value, `reportValidity()` must be called first as described in [Input Validation](/elements/pssword-field#input-validation).
```

See the [Input Length](/elements/password-field#input-length) example below for more detail.

## Input length

The `maxlength` attribute limits the number of characters that can be typed into the input, and the `minlength` attribute sets the minimum of characters. `ef-password-field` will show error styles if a condition is not met.

> @> Constraint validation is only applied when the value is changed by the user. [See input password](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/password#maxlength).

Reviewer: ```suggestion
@> `maxlength` and `minlength` constraint validations are only applied when the value is changed by the user. [See input password](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/password#maxlength).
```

::
```javascript
::password-field::
const passwordField = document.querySelector("ef-password-field");
const errorText = document.getElementById('error-text');

passwordField.addEventListener("blur", () => {
  if (passwordField.error) {
    errorText.innerHTML = "Password length must be between 8 - 16 characters";
  }
  else {
    errorText.innerHTML = "";

>   }

Reviewer: ```suggestion
  errorText.innerHTML = passwordField.error ? "Password length must be between 8 - 16 characters." : "";
```

```javascript
const passwordField = document.querySelector("ef-password-field");
const errorText = document.getElementById('error-text');

passwordField.addEventListener("blur", () => {
  if (passwordField.error) {
    errorText.innerHTML = "Password length must be between 8 - 16 characters";
  }
  else {
    errorText.innerHTML = "";

>   }

Reviewer: ```suggestion
  errorText.innerHTML = passwordField.error ? "Password length must be between 8 - 16 characters." : "";
```

});

passwordField.addEventListener("input", () => {
  if (!passwordField.error) {
    errorText.innerHTML = "";
  }
});
```

```typescript
import { PasswordField } from '@refinitiv-ui/elements/password-field';

> const passwordField = document.querySelector("ef-password-field") as PasswordField;

Reviewer: ```suggestion
const passwordField = document.querySelector<PasswordField>("ef-password-field");
```

> const errorText = document.getElementById('error-text');

Reviewer: double quote consistency :)
```suggestion
const errorText = document.getElementById("error-text");
```

passwordField?.addEventListener("blur", () => {
  if (!errorText) {
    return;
  }
  if (passwordField?.error) {
    errorText.innerHTML = "Password length must be between 8 - 16 characters";
  }
  else {
    errorText.innerHTML = "";

>   }

Reviewer: ```suggestion
  errorText.innerHTML = passwordField.error ? "Password length must be between 8 - 16 characters." : "";
```

::password-field::
const passwordField = document.querySelector("ef-password-field");
const errorText = document.getElementById("error-text");

passwordField.addEventListener("blur", (event) => {
  if (passwordField.error) {
    errorText.innerHTML = "Password is too weak.";
  } else {
    errorText.innerHTML = "";

>   }

Reviewer: ```suggestion
  errorText.innerHTML = passwordField.error ? "Password is too weak." : "";
```

```javascript
const passwordField = document.querySelector("ef-password-field");
const errorText = document.getElementById("error-text");

passwordField.addEventListener("blur", () => {
  if (passwordField.error) {
    errorText.innerHTML = "Password is too weak.";
  } else {
    errorText.innerHTML = "";

>   }

Reviewer: ```suggestion
  errorText.innerHTML = passwordField.error ? "Password is too weak." : "";
```

});

passwordField.addEventListener("input", () => {
  if (!passwordField.error) {
    errorText.innerHTML = "";
  }
});
```

```typescript
import { PasswordField } from '@refinitiv-ui/elements/password-field';

> const passwordField = document.querySelector("ef-password-field") as PasswordField;

Reviewer: ```suggestion
const passwordField = document.querySelector<PasswordField>("ef-password-field");
```

const errorText = document.getElementById("error-text");

passwordField?.addEventListener("blur", () => {
  if (!errorText) {
    return;
  }
  if (passwordField?.error) {
    errorText.innerHTML = "Password is too weak.";
  } else {
    errorText.innerHTML = "";

>   }

Reviewer: let's simplify with this one.
```
  errorText.innerHTML = passwordField.error ? "Password is too weak." : "";
```
